astropy/io/misc/asdf/tags/transform/basic.py

```diff
@@ -130,16 +147,27 @@ def assert_equal(cls, a, b):
 
 class ConstantType(TransformType):
     name = "transform/constant"
-    types = ['astropy.modeling.functional_models.Const1D']
+    version = '1.4.0'
+    types = ['astropy.modeling.functional_models.Const1D',
+             'astropy.modeling.functional_models.Const2D']
 
     @classmethod
     def from_tree_transform(cls, node, ctx):
-        return functional_models.Const1D(node['value'])
+        if node['dimensions'] == 1:
+            return functional_models.Const1D(node['value'])
+        elif node['dimensions'] == 2:
```

Comment:
If `dimensions` is > 2, then this is going to return `None`, which is going to cause confusing errors down the line.  I'd recommend raising an exception right away with a helpful message, something along the lines of "dimensions > 2 is not supported by this library".